test/unit/sz.c

```diff
@@ -0,0 +1,55 @@
+#include "test/jemalloc_test.h"
+
+static size_t
+reg_size_compute(int lg_base, int lg_delta, int ndelta) {
+    return (ZU(1) << lg_base) + (ZU(ndelta) << lg_delta);
+}
+
+TEST_BEGIN(test_sz_psz2ind) {
+    pszind_t zero_ind = sz_psz2ind(0);
+    expect_lu_eq(zero_ind, zero_ind, "sz_psz2ind(0) can't run");
+    for (size_t i = 0; i <= (1 << SC_LG_NGROUP); i++) {
+        for (size_t psz = i * (1 << LG_PAGE) + 1;
+             psz <= (i + 1) * (1 << LG_PAGE); psz++) {
+            pszind_t ind = sz_psz2ind(psz);
+            expect_zu_eq(ind, i, "Got %u as sz_psz2ind of %zu", ind, psz);
+        }
+    }
+
```

Comment:
It would be nice if you could explain what you're testing here and what `base_ind` is. I figured it out, but it took me some time, despite my understanding of the topic.